trace/doc.go

```diff
@@ -46,8 +46,8 @@ lines to the top of the function:
     ctx = trace.StartSpan(ctx, "your choice of name")
     defer trace.EndSpan(ctx)
 
-StartSpan will create a child span if one already exists, and will create a
-new top-level span otherwise.
+Invoking StartSpan with the name of an existing span will create a child span.
```

Comment:
Is this true? It shouldn't have anything to do with the span name. It should create a child span if there is a span in the current context.

Reply:
I'd say, StartSpan will create a new top-level span if the context doesn't contain another span.

Otherwise, it creates a child.

Reply:
Wow, what was I thinking, for some reason I thought I had read through code that reused the parent if there existed a parent in the context but also if the name was the same. Sorry about that, and thank you @rakyll for this correction.